File: 2021 czerwiec/c++/3.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

bool is_pal(const string& s)
{
    for (int i = 0; i < s.size() / 2; i++)
        if (s[i] != s[s.size() - 1 - i])
            return false;
    return true;
}

int main()
{
    ifstream in("napisy.txt");
    
    string s;
    while (in >> s)
    {
        if (is_pal(s + s[0]))
            cout << s[s.size() / 2];
        else if (is_pal(s[s.size() - 1] + s))
            cout << s[s.size() / 2 - 1];
    }
}

File: 2021 czerwiec/python/3.py
with open("napisy.txt", "r") as f:
    file = f.read().strip().split('\n')

def is_pal(s):
    return s == s[::-1]

for s in file:
    if is_pal(s + s[0]):
        print(s[len(s) // 2], end="")
    elif is_pal(s[-1] + s):
        print(s[len(s) // 2 - 1], end="")